SCClassLibrary/Common/Math/Integer.sc

```diff
@@ -171,6 +171,14 @@ Integer : SimpleNumber {
 		*/
 	}
 
+	factorial {
+		if(this > 12) {
+			"Insufficient integer precision for this number (factorial(%)). "
+			"Try to use float instead.".format(this).warn
```

Comment:
why not return a float in this case?

Reply:
and should the same also go for `doubleFactorial`?

Reply:
It was just for backward compatibility. We could also make it an API change, but I thought it might be more polite like this.

Reply:
`doubleFactorial` is new, so we would have to discuss how it should behave.

Reply:
sorry, i mean return the integer only when precision allows, and floating-point otherwise. we have another discussion going on about this below but i just wanted to clarify up here

Reply:
I see. I 'm not so sure if this is really such a good idea. This complicates the explanation quite a bit. Maybe we need to throw an error also for `doubleFactorial`?